homeassistant/components/yale_smart_alarm/alarm_control_panel.py

```diff
@@ -46,66 +53,92 @@
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the alarm platform."""
-    name = config[CONF_NAME]
-    username = config[CONF_USERNAME]
-    password = config[CONF_PASSWORD]
-    area_id = config[CONF_AREA_ID]
-
-    try:
-        client = YaleSmartAlarmClient(username, password, area_id)
-    except AuthenticationError:
-        _LOGGER.error("Authentication failed. Check credentials")
-        return
-
-    add_entities([YaleAlarmDevice(name, client)], True)
-
-
-class YaleAlarmDevice(AlarmControlPanelEntity):
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Import Yale configuration from YAML."""
+    LOGGER.warning(
+        "Loading Yale Alarm via platform setup is deprecated; Please remove it from your configuration"
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the alarm entry."""
+
+    async_add_entities(
+        [YaleAlarmDevice(coordinator=hass.data[DOMAIN][entry.entry_id]["coordinator"])]
+    )
+
+
+class YaleAlarmDevice(CoordinatorEntity, AlarmControlPanelEntity):
     """Represent a Yale Smart Alarm."""
 
-    def __init__(self, name, client):
-        """Initialize the Yale Alarm Device."""
-        self._name = name
-        self._client = client
-        self._state = None
+    coordinator: YaleDataUpdateCoordinator
 
-        self._state_map = {
-            YALE_STATE_DISARM: STATE_ALARM_DISARMED,
-            YALE_STATE_ARM_PARTIAL: STATE_ALARM_ARMED_HOME,
-            YALE_STATE_ARM_FULL: STATE_ALARM_ARMED_AWAY,
-        }
+    _attr_name = coordinator.entry.data[CONF_NAME]
+    _attr_entry_id = coordinator.entry.entry_id
+    _attr_identifier = coordinator.entry.data[CONF_USERNAME]
 
     @property
     def name(self):
         """Return the name of the device."""
-        return self._name
+        return str(self._attr_name)
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this entity."""
+        return self._attr_entry_id
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device information about this entity."""
+        return {
+            ATTR_NAME: str(self._attr_name),
```

Comment:
```suggestion
            ATTR_NAME: self.name,
```

Reply:
I did have to wrap self.name into str() or mypy was complaining on me:
Incompatible types (expression has type "Optional[str]", TypedDict item "name" has type "str")  [typeddict-item]
Not sure there is an option here to do write something else?